Source/WebCore/dom/Document.cpp

```diff
@@ -3170,6 +3170,14 @@ bool Document::updateLayoutIfDimensionsOutOfDate(Element& element, OptionSet<Dim
         // Check our containing block chain. If anything in the chain needs a layout, then require a full layout.
         for (CheckedPtr currentRenderer = renderer; currentRenderer && !currentRenderer->isRenderView(); currentRenderer = currentRenderer->container()) {
 
+            if (currentRenderer->isSVGRenderer()) {
```

Comment:
Instead of checking every renderer in the block chain if it `isSVGRenderer()`, I think this check can be moved inside the if-statement below:
```
    if (!currentRendererBox) {
        if (currentRenderer->isSVGRenderer()) {
            ...
        } else {
            requireFullLayout = true;
            break;
        }
    }
```

Reply:
This doesn't quite work unfortunately, since some SVG renderers (LegacyRenderSVGRoot) is also a box, and we want to use the SVG behaviour.